bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/util/DynamicContextBuilder.java

```diff
@@ -144,9 +146,22 @@ public URI resolveUri(String uri) {
 			if (realURI.isAbsolute()) {
 				return realURI;
 			}
+
+			if (_staticContext.getBaseUri() == null) {
+				throw DynamicError.noBaseURI();
```

Comment:
It's strange for there to not be a base URI, because typically you are processing a source document from which you can derive the base URI...but I guess it could happen that the source document isn't associated with a URI, because it was generated on the fly for example....

That said I'd agree that FONS0005 fits in this case.